domain/personallearningplan/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/personallearningplan/service/GoalPersistenceAdapter.kt

```diff
@@ -29,4 +31,14 @@ interface GoalPersistenceAdapter {
    * Updates a [Goal] identified by its `prisonNumber` and [UpdateGoalDto.reference]
    */
   fun updateGoal(prisonNumber: String, updatedGoalDto: UpdateGoalDto): Goal?
+
+  /**
+   * Archives a [Goal] identified by its `prisonNumber` and [ArchiveGoalDTO.reference]
+   */
+  fun archiveGoal(prisonNumber: String, archiveGoalDto: ArchiveGoalDto): Goal?
```

Comment:
I feel like we should add prison number to the DTO as it's required to make the request/identify the goal